### PR TITLE
Add section about new Debug Page to Core Concepts > Cypress App

### LINKED ISSUE #4924: Document new Debug Page on Cypress App page
### Subject

Guides -> Core Concepts -> Cypress App

### Description

Need a para or two explaining the functionality of the new Debug Page.

### PR DESCRIPTION
**Pull request by debrisapron**
### DO NOT MERGE UNTIL DEBUG RELEASE (01/31)

Closes #4924

### CODE REVIEW

**Comment by debrisapron** [comment]
Preview content here: https://cypress-documentation-4lf0wisd9-cypress-io.vercel.app/guides/core-concepts/cypress-app#The-Debug-Page

File: content/guides/core-concepts/cypress-app.md [hunk 1]
```diff
@@ -159,6 +159,22 @@ The title of each run is taken from the
 for that change, and clicking on it will take you to the corresponding run page
 in Cypress Cloud.
 
+### The Debug Page
+
+<DocsImage src="/img/guides/core-concepts/cypress-app/debug-page.png" alt="The Debug Page"></DocsImage>
+
+Like the Project Runs page above, the Debug Page surfaces data from Cypress
+Cloud (see [note above](#Overview)) directly inside Cypress App. However, this
```
Comment (Jess-Cypress) [inline content/guides/core-concepts/cypress-app.md:167]: Per the style guide (https://docs.google.com/document/d/1pOZ0Ntm4b1Eev_iY5vLBPhOxx66Z8Njrvvg6WMCvWRI/edit), this "App" shouldn't be capitalized: the Cypress app.

**Comment by mattvCypress** [comment]
@pstakoun any objections to dropping "The" in these places?

image

**Comment by pstakoun** [comment]
> @pstakoun any objections to dropping "The" in these places?
> 
> image

@mattvCypress no objection. Just the headers? Or anywhere else including body?

**Comment by mattvCypress** [comment]
> > @pstakoun any objections to dropping "The" in these places?
> > image
> 
> @mattvCypress no objection. Just the headers? Or anywhere else including body?

I'd just change it in the nav and headers -- it sticks out like a sore thumb there, but it reads more naturally in the paragraph(s).

File: docs/guides/core-concepts/cypress-app.mdx [hunk 2]
```diff
@@ -172,6 +172,36 @@ The title of each run is taken from the
...
+
+This feature eliminates the tedious process of switching between Cypress, your
+local editor, and CI output, in order to fix failed tests. It does this by
+enabling you to use the [Test Runner](#The-Test-Runner) to run only the tests
+that failed in your last recorded test run, as well as review screenshots,
+videos, and logs from your tests.
+
+The Debug page shows the latest completed [test run](/guides/cloud/runs) that
+matches your current locally checked out git commit. If no run is found for your
+current commit, then Cypress crawls your git tree to find the most recent run in
```
Comment (tbiethman) [inline docs/guides/core-concepts/cypress-app.mdx:195]: While not inaccurate, `Cypress crawls your git tree` might sound a little more invasive than we'd like? Could just describe the end result here:

```suggestion
current commit, then Cypress shows the most recent run in
```

File: docs/guides/core-concepts/cypress-app.mdx [hunk 3]
```diff
@@ -172,6 +172,36 @@ The title of each run is taken from the
...
+without leaving Cypress.
+
+This feature eliminates the tedious process of switching between Cypress, your
+local editor, and CI output, in order to fix failed tests. It does this by
+enabling you to use the [Test Runner](#The-Test-Runner) to run only the tests
+that failed in your last recorded test run, as well as review screenshots,
+videos, and logs from your tests.
+
+The Debug page shows the latest completed [test run](/guides/cloud/runs) that
+matches your current locally checked out git commit. If no run is found for your
```
Comment (tbiethman) [inline docs/guides/core-concepts/cypress-app.mdx:194]: I know we've gone around a bit on the terminology here. I'll throw out an alternative but no strong feelings either way:

```suggestion
matches the HEAD commit of your local project. If no run is found for your
```

File: docs/guides/core-concepts/cypress-app.mdx [hunk 4]
```diff
@@ -172,6 +172,36 @@ The title of each run is taken from the
...
+current commit, then Cypress crawls your git tree to find the most recent run in
+your branch's history, including parent branches. The first 100 failed tests are
+listed, with a link to Cypress Cloud to review the full run.
+
+To use this feature, you must ensure that your version control is properly
+configured and that you are sending git information to Cypress Cloud. Under some
+environment setups (e.g. `docker`/`docker-compose`), you may have to pass git
+information using
+[custom environment variables](/guides/continuous-integration/introduction#Git-information).
```
Comment (tbiethman) [inline docs/guides/core-concepts/cypress-app.mdx:203]: Do we want to add an explicit note that other version control systems aren't supported? Or do we assert that elsewhere already?

File: docs/guides/core-concepts/cypress-app.mdx [hunk 5]
```diff
@@ -172,6 +172,36 @@ The title of each run is taken from the
...
+This feature eliminates the tedious process of switching between Cypress, your
+local editor, and CI output, in order to fix failed tests. It does this by
+enabling you to use the [Test Runner](#The-Test-Runner) to run only the tests
+that failed in your last recorded test run, as well as review screenshots,
+videos, and logs from your tests.
+
+The Debug page shows the latest completed [test run](/guides/cloud/runs) that
+matches your current locally checked out git commit. If no run is found for your
+current commit, then Cypress crawls your git tree to find the most recent run in
+your branch's history, including parent branches. The first 100 failed tests are
```
Comment (tbiethman) [inline docs/guides/core-concepts/cypress-app.mdx:196]: I think we could remove "including parent branches" here, as there's not really a parent/child relationship between branches in git. Once you label a commit with a branch, all history up until that point is considered part of that branch.

File: docs/guides/cloud/runs.mdx [hunk 6]
```diff
@@ -340,12 +340,22 @@ Cypress Cloud.
...
 <DocsImage
   src="/img/guides/core-concepts/cypress-app/spec-explorer.png"
-  alt="The Spec Explorer"
+  alt="Specs"
+/>
+
+The [Debug](/guides/core-concepts/cypress-app#Debug) page allows you to easily
+debug failed CI test runs from within the Cypress app. From this page, you can
+review, rerun, and debug failed CI test runs that are recorded to Cypress Cloud
+– all without leaving Cypress.
```
Comment (Jess-Cypress) [inline docs/guides/cloud/runs.mdx:356]: all without leaving the Cypress app.

File: docs/guides/core-concepts/cypress-app.mdx (hunk 4 above)
Comment (pstakoun) [reply docs/guides/core-concepts/cypress-app.mdx:203]: We typically only support git for all our features. I made it a bit more explicit by changing "version control" to "git" up above though.

Comment (pstakoun) [reply docs/guides/core-concepts/cypress-app.mdx:203]: We should probably make this more explicit throughout our docs though, this is a good callout. cc @ryanpei

**Review by tbiethman — DISMISSED** [review]
All of my feedback has been addressed, looking good to me!

**Review by pstakoun — DISMISSED** [review]

File: docs/guides/component-testing/faq.mdx [hunk 7]
```diff
@@ -106,7 +106,7 @@ component, just like in the application. See our guide on
 [Styling Components](/guides/component-testing/styling-components) for more
 information on doing so.
 
-### Why doesn't my spec show in the spec explorer?
+### Why doesn't my spec show in the Specs page?
```
Comment (mattvCypress) [inline docs/guides/component-testing/faq.mdx:109]: @pstakoun Did you already do a CMD+F for other instances of "spec(s) explorer"? I don't think there were many.

Comment (warrensplayer) [reply docs/guides/component-testing/faq.mdx:109]: I just did a check and the only things I could find where the name of image files or in the change log.

image

**Review by warrensplayer — APPROVED** [review]

**Comment by rockhold** [comment]
The Debug page example image is shown using Cy10.8.0. Any desire to get a screenshot with a newer version?

**Review by rockhold — APPROVED** [review]
Docs LGTM!

File: docs/guides/component-testing/faq.mdx (hunk 7 above)
Comment (pstakoun) [reply docs/guides/component-testing/faq.mdx:109]: Yeah I checked as well and caught a couple, didn't see any more

**Comment by pstakoun** [comment]
@rockhold good callout re:the screenshot – I've updated it. Thanks!

**Review by pstakoun — APPROVED** [review]

**Review by emilyrohrbough — COMMENTED** [review]
Will there be a follow up PR to add this content to the Debugging guide as well?
https://docs.cypress.io/guides/guides/debugging

**Comment by pstakoun** [comment]
> Will there be a follow up PR to add this content to the Debugging guide as well? https://docs.cypress.io/guides/guides/debugging

This is a great idea and was not previously considered.

**Pull request merged**